docs/sources/next/using-k6/javascript-compatibility-mode.md

```diff
@@ -1,44 +1,71 @@
 ---
-title: JavaScript Compatibility Mode
+title: JavaScript compatibility mode
+menuTitle: JavaScript mode
 excerpt: 'k6 supports running test scripts with different ECMAScript compatibility modes using --compatibility-mode'
 _build:
   list: false
 weight: 19
 ---
 
-# JavaScript Compatibility Mode
+# JavaScript compatibility mode
 
-You can run test scripts with different ECMAScript compatibility modes with the
-`run --compatibility-mode` CLI option or `K6_COMPATIBILITY_MODE` environment variable.
+You can write k6 scripts in various ECMAScript versions:
 
-Currently two modes are available:
+- ES6+ JavaScript with ES modules (ESM).
+- Plain old JavaScript (ES5.1+) with CommonJS modules.
 
-## Base
+k6 supports both module types and most ES6+ features in all k6 execution modes: local, distributed, and cloud.
+
+Yet, to enable ES module support, k6 uses [Babel](https://babeljs.io/) internally to transform ESM to CommonJS. This process is as follows:
+
+![Babel transformation in k6](/media/docs/k6-oss/diagram-grafana-k6-babel-pipeline.png)
+
+The k6 Babel transformation may slightly delay the startup for executing large scripts. Thus, some users prefer to bundle their test code outside k6. For this reason, k6 offers two JavaScript Compatibility modes:
```

Comment:
```suggestion
When executing large scripts, the k6 Babel transformation can slightly delay the startup time. Because of that, some users prefer to bundle their test code outside k6. For this reason, k6 offers two JavaScript compatibility modes:
```